lib/project_types/script/layers/infrastructure/assemblyscript_dependency_manager.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+module Script
+  module Layers
+    module Infrastructure
+      class AssemblyScriptDependencyManager
+        def initialize(ctx, language, extension_point, script_name)
+          @ctx = ctx
+          @language = language
+          @script_name = script_name
+          @extension_point = extension_point
+        end
+
+        def bootstrap
+          write_npmrc
+          write_package_json
+        end
+
+        def installed?
+          # Assuming if node_modules folder exist at root of script folder, all deps are installed
+          Dir.exist?("node_modules")
+        end
+
+        def install
+          output, status = @ctx.capture2e("npm", "install", "--no-audit", "--no-optional", "--loglevel error")
+          raise DependencyInstallError, output unless status.success?
+        end
+
+        private
+
+        def write_npmrc
+          npmrc = <<~HERE
+            @shopify:registry=https://registry.npmjs.org/
+          HERE
```

Comment:
An alternative way to do this would be
```
@ctx.system('npm', '--userconfig', './.npmrc', 'config', 'set', '@shopify:registry', 'https://registry.npmjs.com')
```
within the project directory.  By using the `npm` command itself to set the `.npmrc` file, it future-proofs the code against any changes to the format (however unlikely).